### PR TITLE
Switching to use generations tfms

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -1258,7 +1258,14 @@ functions @{
                 {
                     continue;
                 }
-                projectAssemblies.Add(Path.Combine(Directory.GetCurrentDirectory(), "artifacts", "build", name, target.Framework, name + ".dll"));
+                var path = Path.Combine(Directory.GetCurrentDirectory(), "artifacts", "build", name, target.Framework, name + ".dll");
+                if (!File.Exists(path))
+                {
+                    var directoryName = target.Framework == "dnx451" ? "net451" : "dotnet5.4";
```
Comment (pranavkm) [inline makefile.shade]: For the newly changed Abstractions projects

**Comment by pranavkm** [comment]
Checking this in to keep CI going. I'll keep the PR open

File: src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs [hunk 2]
```diff
@@ -99,7 +99,7 @@ private static CSharpCompilationOptions AddSigningOptions(CSharpCompilationOptio
 
             if (!string.IsNullOrEmpty(keyFile))
             {
-#if DNXCORE50
+#if DOTNET5_45_45_4
```
Comment (davidfowl) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: Wtf is this :smile:

File: makefile.shade (hunk 1 above)
Comment (davidfowl) [inline makefile.shade]: :smile:

File: src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs (hunk 2 above)
Comment (anurse) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: Looks like find/replace went a little haywire ;P

Comment (pranavkm) [inline src/Microsoft.Dnx.Compilation.CSharp.Common/CompilerOptionsExtensions.cs]: Fixed this before pushing.

**Pull request closed without merging**